### PR TITLE
feat(themes): add component level tokens for themeing

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #112

## Summary:

All token work is updated to support the new component theme infrastructure.

`advisory` was removed as part of the design switch to `warning` only.

The additional variants were added.

varient examples were updated to better show each varient in a recognizable way

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

## Verification

- [ ] No linter issues
- [ ] All automated tests pass
- [ ] New Tier 3 component token exists
- [ ] Advisory variant is removed
- [ ] API Page "Theme" section is valid

DEFAULT
- [ ] `default` variant renders correctly
- [ ] `accent` variant renders correctly 
- [ ] `disabled` variant renders correctly
- [ ] `emphasis` variant renders correctly
- [ ] `error` variant renders correctly
- [ ] `info` variant renders correctly
- [ ] `secondary` variant renders correctly
- [ ] `subtle` variant renders correctly
- [ ] `success` variant renders correctly
- [ ] `tertiary` variant renders correctly
- [ ] `warning` variant renders correctly

INVERSE
- [ ] `default` variant renders correctly
- [ ] `accent` variant renders correctly 
- [ ] `disabled` variant renders correctly
- [ ] `emphasis` variant renders correctly
- [ ] `error` variant renders correctly
- [ ] `info` variant renders correctly
- [ ] `secondary` variant renders correctly
- [ ] `subtle` variant renders correctly
- [ ] `success` variant renders correctly
- [ ] `tertiary` variant renders correctly
- [ ] `warning` variant renders correctly

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/tokens.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+:host {
+  --ds-auro-icon-color: var(--ds-color-icon-primary-default, $ds-color-icon-primary-default);
```
Comment (jordanjones243) [inline src/tokens.scss:2]: Is this still needed? I see the same line in the `styleColor.scss` file

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

File: src/tokens.scss (hunk 1 above)
Comment (jason-capsule42) [reply src/tokens.scss:2]: It was needed here and should have been removed from `styleColor.scss` file. This is updated.

File: index.js [hunk 2]
```diff
@@ -6,10 +6,12 @@ import { AuroIcon } from './src/auro-icon.js';
  * @returns {void}
  */
  const registerComponent = (name = 'custom-icon') => {
+  console.warn('registerComponent', name);
```
Comment (jordanjones243) [inline index.js:9]: Remove console warnings

File: apiExamples/visualState.html [hunk 3]
```diff
@@ -1,7 +1,13 @@
-<auro-icon category="interface" name="pin-trip"></auro-icon>
-<auro-icon category="interface" name="pin-trip" emphasis></auro-icon>
-<auro-icon category="interface" name="pin-trip" accent></auro-icon>
-<auro-icon category="interface" name="pin-trip" disabled></auro-icon>
-<auro-icon category="interface" name="pin-trip" error></auro-icon>
-<auro-icon category="interface" name="pin-trip" success></auro-icon>
-<auro-icon category="interface" name="pin-trip" advisory></auro-icon>
+<auro-icon category="logos" name="tail-AS"></auro-icon> default<br />
```
Comment (jordanjones243) [inline apiExamples/visualState.html:1]: Looks like this icon is broken in the example

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

**Pull request merged**